src/tools/android/java/com/google/devtools/build/android/desugar/LambdaDesugaring.java

```diff
@@ -473,7 +513,28 @@ public void visitInvokeDynamicInsn(String name, String desc, Handle bsm, Object.
                 + " with arguments "
                 + Arrays.toString(bsmArgs),
             e);
+      } catch (IllegalStateException e) {
+        // Provide the instructions to help understanding precondition checks.
+        // This is an unexpected failure state, so the extra overhead is justified.
+        Printer p = new Textifier();
+        TraceMethodVisitor visitor = new TraceMethodVisitor(p);
+        for (AbstractInsnNode n : instructions.toArray()) {
+          n.accept(visitor);
+        }
+        StringWriter b = new StringWriter();
+        p.print(new PrintWriter(b));
+        throw new IllegalStateException("Unexpected instructions in :\n" + b.toString(), e);
+      }
+    }
+
+    /** If node is not an instruction, scans backwards until it finds one that is. */
+    private AbstractInsnNode findInstruction(AbstractInsnNode node) {
```

Comment:
is skipping over labels really a good idea?  seems like if there's branches then rewriting code before and after an incoming jump destination could be problematic, not to mention that we may not even be looking at code that will ever executed in the order that it appears (in the case of unconditional jumps)